zebra-network/src/peer_set/candidate_set.rs

```diff
@@ -341,12 +345,24 @@ fn validate_addrs(
 
     // TODO:
     // We should eventually implement these checks in this function:
-    // - Zebra should stop believing far-future last_seen times from peers (#1871)
     // - Zebra should ignore peers that are older than 3 weeks (part of #1865)
     //   - Zebra should count back 3 weeks from the newest peer timestamp sent
     //     by the other peer, to compensate for clock skew
     // - Zebra should limit the number of addresses it uses from a single Addrs
     //   response (#1869)
 
-    addrs
+    let addrs: Vec<_> = addrs.into_iter().collect();
+
+    let most_recent_reported_seen_time = addrs
+        .iter()
+        .map(|addr| addr.get_last_seen())
+        .max()
+        .unwrap_or(last_seen_limit);
```

Comment:
Suggestion: We could return early if the response is empty, and replace this default with an `expect`.

That way, we don't have to make sure that the default always does the right thing. (In this case, it's fine, because the default ends up calculating zeroes and operating on empty iterators. But I needed to read the code a few times to check that.)